### ISSUE TITLE
Enhancement: New Compiler Possible Collision Problem - Refactor the DbDataReader GetHashCode() Generator

### ISSUE THREAD

**Issue by mikependon** [mikependon/RepoDB#597]
Only in the new compiler. In order for us to ensure that the collision will not happen on the caching of the AOT Compiled Expression, we need to consider the caching of the following.

- HashCode of the Type
- HashCode of the Name of each DbDataReader Fields
- HashCode of the Type of each DbDataReader Fields
- HashCode of the Ordinal of each DbDataReader Fields <-- this is not yet there

Without the ordinal caching, the collision may happen ended-up multiple invalid casting exception (and/or data assignment).

```csharp
connection.ExecuteQuery<Person>("SELECT Id, Name FROM Person;");
connection.ExecuteQuery<Person>("SELECT Name, Id FROM Person;");
```

The possible fix is very minimal, simply add the concatenation of the ordering.

```csharp
private static long GetReaderFieldsHashCode(DbDataReader reader)
{
	if (reader == null)
	{
		return 0;
	}
	var hashCode = (long)0;
	for (var ordinal = 0; ordinal < reader.FieldCount; ordinal++)
	{
		// The spatial data type is null.
		hashCode += string.Concat(reader.GetName(ordinal), "-", ordinal).GetHashCode() +
			(reader.GetFieldType(ordinal)?.GetHashCode()).GetValueOrDefault();
	}
	return hashCode;
}
```

**Comment by mikependon** [comment]
Linking this fix. [f992f61](https://github.com/mikependon/RepoDB/commit/0e86b5e5345bfe16f470b04fa24d310b4558a291)

**Comment by mikependon** [comment]
This has been fixed and is now available at [RepoDb v1.12.3](https://www.nuget.org/packages/RepoDb/1.12.3). Closing this incident ticket now.
